src/backend/vulkan/src/lib.rs

```diff
@@ -432,6 +430,44 @@ impl hal::PhysicalDevice<Backend> for PhysicalDevice {
         }
     }
 
+    fn image_format_properties(
+        &self, format: format::Format, dimensions: u8, tiling: image::Tiling,
+        usage: image::Usage, storage_flags: image::StorageFlags,
+    ) -> Option<image::FormatProperties> {
+        match self.instance.0
+            .get_physical_device_image_format_properties(
+                self.handle,
+                conv::map_format(format),
+                match dimensions {
+                    1 => vk::ImageType::Type1d,
+                    2 => vk::ImageType::Type2d,
+                    3 => vk::ImageType::Type3d,
+                    _ => panic!("Unexpected image dimensionality: {}", dimensions)
+                },
+                conv::map_tiling(tiling),
+                conv::map_image_usage(usage),
+                conv::map_image_flags(storage_flags),
+            )
+        {
+            Ok(props) => Some(image::FormatProperties {
+                max_extent: image::Extent {
+                    width: props.max_extent.width,
+                    height: props.max_extent.height,
+                    depth: props.max_extent.depth,
+                },
+                max_levels: props.max_mip_levels as _,
+                max_layers: props.max_array_layers as _,
+                sample_count_mask: props.sample_counts.flags() as _,
```

Comment:
This kind of mismatches with the sample bit mask used in dx12 and metal:
1-bit sample will return 1 instead of 2.